chore: improve contrib guide with e2e test details (#447)
open-feature/go-sdk-contrib: CONTRIBUTING.md

## System Requirements

> go 1.19+  is recommended.

Reviewer: Isn't [it](https://github.com/open-feature/go-sdk-contrib/blob/main/providers/go-feature-flag/go.mod) incompatible with go 1.19? As [toolchain](https://go.dev/doc/toolchain) requiers go 1.21+.

Author: Yes, some providers are already on 1.21 and this here is for guidance only. Our pipelines on the other hand runs with  latest go version

Reviewer: Yeah, Okay

## Setup workspace
 

> Run the following command to set up the go workspace,

Reviewer 2: ```suggestion
Run the following command to set up the go workspace:
```

> ```

Reviewer 2: ```suggestion
```shell
```

make workspace-init
```

> To sync go.work with current modules,

Reviewer 2: ```suggestion
To sync `go.work` with the current modules, run:
```

```shell
make workspace-update
```

## Adding a module

The project provides `makefile` targets for creating [hooks](https://openfeature.dev/docs/reference/concepts/hooks) and [providers](https://openfeature.dev/docs/reference/concepts/provider).
To contribute a new hook or provider, fork this repository and create a new go module, it will then be discoverable by `make workspace-init` and `make workspace-update`.

> To automatically create and setup new provider directory  use following command (requires jq),

Reviewer 2: ```suggestion
To automatically create and set up a new provider directory, use the following command (requires [jq](https://jqlang.github.io/jq/)):
```

```shell
make MODULE_NAME=NAME new-provider
```

> To automatically create and setup new hook directory, use following command (requires jq),

Reviewer 2: ```suggestion
To automatically create and set up a new hook directory, use the following command (requires [jq](https://jqlang.github.io/jq/)):
```

Any published modules must have reasonable test coverage.

> To run tests in all existing go modules, use command,

Reviewer 2: ```suggestion
To run tests in all existing go modules, use the command:
```

```shell
make test
```

It is recommended to include end-to-end (e2e) tests in your provider when possible. 
If you have dependency services for your e2e tests, make sure to add them as service in the build pipeline.
Also, make sure to include them in the makefile for easy use by contributors.

> To start e2e test helpers use command,

Reviewer 2: ```suggestion
To start e2e test helpers use the command:
```

```shell
make e2e-start-helpers
```

> Then, you can run all tests including e2e tests using command,

Reviewer 2: ```suggestion
Then, you can run all tests, including e2e tests using the command:
```

```shell
make e2e
```

> Once tests are complete, you can remove e2e test helpers using command,

Reviewer 2: ```suggestion
Once tests are complete, you can remove e2e test helpers using the command:
```
